### PR TITLE
Added 'Orientation' parameter for vertical and horizontal layouts

### PR DESCRIPTION
**Pull request by fhaefele**
test_horizontal
example
example2

### CODE REVIEW

File: Violin.m [hunk 1]
```diff
@@ -1,714 +1,726 @@
...
+                        case 'right'
+                            [x,y] = obj.plotOrientation([pos-((counts')/max(counts))*max(jitterstrength)*2, pos*ones(size(counts,2),1)]',...
+                                        [edges(1:end-1)+max(diff(edges))/2; edges(1:end-1)+max(diff(edges))/2]);    
+                        case 'left'
+                            [x,y] = obj.plotOrientation([pos*ones(size(counts,2),1), pos+((counts')/max(counts))*max(jitterstrength)*2]', ...
+                                        [edges(1:end-1)+max(diff(edges))/2; edges(1:end-1)+max(diff(edges))/2]);
+                        otherwise
+                            fprintf([namefun, ' No histogram/bar plot option available for full violins, as it would look overcrowded.\n'])
+                    end
+                    obj.HistogramPlot = plot(x,y,'-','LineWidth',1, 'Color', 'k');
```
Comment (bastibe) [inline Violin.m:271]: This is referring to an earlier comment.

Here, the `case`s no longer plot, but merely prepare an `x` and `y`. This makes it harder to understand what's going on. I'd prefer the plot commands to still be inlined, as this is explicitly what the code does: It plots things. Preparing data is merely an ancillary complexity for the dual-orientation use case.

This is exacerbated by the name `plotOrientation`, which implies that the function actually plots. Which it doesn't. It merely swaps two variables.

I'd prefer a structure like this in every `case` instead:

```
x = [pos*ones(size(counts,2),1), pos+((counts')/max(counts))*max(jitterstrength)*2]';
y = [edges(1:end-1)+max(diff(edges))/2; edges(1:end-1)+max(diff(edges))/2];
[x, y] = obj.swapOrientationsMaybe(x, y);
obj.HistogramPlot = plot(x, y, '-', 'LineWidth', 1, 'Color', 'k');
```

Our first objective here is readability and maintainability!

**Review by bastibe — CHANGES_REQUESTED** [review]
Thank you for the contribution! This is super useful!

However, I don't like the way the `plotOrientation` function makes the code harder to reason about in some places. I have sketched out an alternative structure that I'd prefer.

Nevertheless, great work, and thank you very much!

File: Violin.m [hunk 2]
```diff
@@ -1,714 +1,726 @@
...
+        BoxPlot         % fill plot of the box between the quartiles
+        WhiskerPlot     % line plot between the whisker ends
+        MedianPlot      % scatter plot of the median (one point)
+        NotchPlots      % scatter plots for the notch indicators
+        MeanPlot        % line plot of the mean (horizontal line)
+        HistogramPlot   % histogram of the data
+        ViolinPlotQ     % fill plot of the Quartiles as shadow
+    end
+
+    properties (Hidden = true)
```
[the remark this answers is not in the document]
Comment (fhaefele) [reply Violin.m:90]: Sounds good. I will fix it.

File: Violin.m (hunk 1 above)
Comment (bastibe) [reply Violin.m:271]: I would use the above function name `swapOrientationsMaybe`. But feel free to use a different name if you find one.

**Comment by fhaefele** [comment]
Hm, merging and resolving conflicts is still pretty new to me. I gave it my best. Also I have addressed all of your comments. Let me know if everything fits now, else I will try to fix it asap. I was also thinking of including that 'parent' property which you discussed in the other PR.

File: Violin.m [hunk 3]
```diff
@@ -1,714 +1,733 @@
...
+            validDataStyles = {'scatter', 'histogram', 'none'};
+            checkStyle = @(x)any(validatestring(x, validDataStyles));
+            p.addParameter('DataStyle', 'scatter', checkStyle);
+            p.addParameter('Orientation', 'vertical', @(x) ismember(x, {'vertical', 'horizontal'}));
+
+            p.parse(data, pos, varargin{:});
+            results = p.Results;
+        end
+
+        function [x,y] = swapOrientationMaybe(obj, x, y)
```
Comment (bastibe) [inline Violin.m:701]: could you add a quick documentation to the function?

**Review by bastibe — CHANGES_REQUESTED** [review]
This looks much better, thank you!

Just one more quick note: Could you format the code with spaces after operators (such as commas), so it matches the rest of the code?

Otherwise, this looks good to me!

**Comment by fhaefele** [comment]
Should be done and ready to merge. Do you want to also add the features from #24 ? Should be an easy inclusion.

**Comment by bastibe** [comment]
> Do you want to also add the 'parent' property from https://github.com/bastibe/Violinplot-Matlab/pull/24 ?

That would be a good idea, thank you!

**Comment by fhaefele** [comment]
I have added now the parent property and checked the functionality of the orientation swap and the 'parent' axis in a modified `testviolinplot.m`. To me it looks like it works fine.

**Comment by bastibe** [comment]
Looks good to me! Thank you! 

If you want to, you could add an example to the README.

Is this ready to merge on your end?

**Comment by bastibe** [comment]
I'll leave it up to you whether you want to add it to the README.

**Comment by fhaefele** [comment]
I have edited the README and the `readme_figures.m` respectively. The README now shows a code snippet that can work on its own, which the previous one couldn't. I have also added two new test cases which test the 'Parent' and 'Orientation' property. Feel free to merge it into the main branch as you like.

**Comment by bastibe** [comment]
Beautiful! Thank you!

**Pull request merged**